packages/devtools/lib/src/timeline/timeline.dart

```diff
@@ -224,4 +228,85 @@ class TimelineScreen extends Screen {
       timelineController.pause();
     }
   }
+
+  /// Adds a button to the timeline that will dump debug information to text
+  /// files and download them. This will only appear if the [debugTimeline] flag
+  /// is true.
+  void _maybeAddDebugDumpButton() {
+    if (debugTimeline) {
+      upperButtonSection.add(PButton('Debug dump')
+        ..small()
+        ..click(() {
+          // TODO(kenzie): we can replace this with something more sophisticated
+          // in the future, but for now this is a good debugging addition.
+
+          // Trace events in the order we received them.
+          final debugTraceEventsOutput = html.document.createElement('a');
+          debugTraceEventsOutput.setAttribute(
+              'href',
+              html.Url.createObjectUrl(
+                  html.Blob([debugTraceEvents.toString()])));
+          debugTraceEventsOutput.setAttribute('download', 'trace_output.txt');
+          debugTraceEventsOutput.style.display = 'none';
+          html.document.body.append(debugTraceEventsOutput);
+          debugTraceEventsOutput.click();
+          debugTraceEventsOutput.remove();
+
+          // Significant events in the frame tracking process.
+          final debugFrameTrackingOutput = html.document.createElement('a');
```

Comment:
make this a util helper function. I'm sure this is not the last time we will want to download large files from the tool.